x-pack/plugins/ml/public/maps/util.ts

```diff
@@ -109,6 +109,34 @@ export function getInitialAnomaliesLayers(jobId: string) {
   return initialLayers;
 }
 
+export function getInitialSourceIndexFieldLayer(sourceIndicesWithGeoFields: {
+  [key: string]: { geoFields: string[]; dataViewId: string };
+}) {
+  const initialLayers: any = [];
+  for (const index in sourceIndicesWithGeoFields) {
+    if (sourceIndicesWithGeoFields.hasOwnProperty(index)) {
+      const { dataViewId, geoFields } = sourceIndicesWithGeoFields[index];
+
+      geoFields.forEach((geoField) => {
+        initialLayers.push({
```

Comment:
Is there any (easy) way of defining a different color for each layer?

<img width="305" alt="image" src="https://user-images.githubusercontent.com/7405507/186406259-cc48d757-bd3a-4918-8f7a-720c1a98c9e4.png">

Reply:
If you want to specify layer color, populate `style` key in the layer descriptor

```
layerDescriptor.style = VectorStyle.createDescriptor({
  [VECTOR_STYLES.FILL_COLOR]: {
    type: STYLE_TYPE.STATIC,
    options: {
      color: '#FFF',
    },
  },
});
```